server/src/main/java/org/elasticsearch/index/mapper/NumberFieldMapper.java

```diff
@@ -117,6 +117,9 @@ public Builder(String name, NumberType type, ScriptCompiler compiler, boolean ig
                 if (s != null && multiFieldsBuilder.hasMultiFields()) {
                     throw new MapperParsingException("Cannot define multifields on a field with a script");
                 }
+                if (s != null && copyTo.hasValues()) {
+                    throw new MapperParsingException("Cannot define copy_to parameter on a field with a script");
+                }
```

Comment:
multi fields and copy to are passed down to the base builder class right? would it be worth to make that more script-aware so that it can throw error, and we can centralize this error handling?

Reply:
++, I've added an `addScriptValidation` method to FieldMapper.Builder and used that to centralise validation.